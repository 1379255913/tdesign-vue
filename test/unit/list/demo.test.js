import { mount } from '@vue/test-utils';
import base from '@/examples/list/demos/base.vue';
import extra from '@/examples/list/demos/extra.vue';
import headerFooter from '@/examples/list/demos/headerFooter.vue';
import imageText from '@/examples/list/demos/imageText.vue';
import multiline from '@/examples/list/demos/multiline.vue';
import operation from '@/examples/list/demos/operation.vue';
import scroll from '@/examples/list/demos/scroll.vue';
import size from '@/examples/list/demos/size.vue';

describe('Button', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('extra demo works fine', () => {
    const wrapper = mount(extra);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('headerFooter demo works fine', () => {
    const wrapper = mount(headerFooter);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('imageText demo works fine', () => {
    const wrapper = mount(imageText);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('multiline demo works fine', () => {
    const wrapper = mount(multiline);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('operation demo works fine', () => {
    const wrapper = mount(operation);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('scroll demo works fine', () => {
    const wrapper = mount(scroll);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
});
