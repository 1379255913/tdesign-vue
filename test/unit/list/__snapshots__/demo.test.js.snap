// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button headerFooter demo works fine 1`] = `
<div>
  <div
    class="t-list t-size-m t-list--split"
  >
    <div
      class="t-list__header"
    >
      这里是 Header
    </div>
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="t-list__footer"
    >
      这里是 Footer
    </div>
    <div
      class="t-list__load"
    />
  </div>
   
  <div
    style="margin-bottom: 16px;"
  />
   
  <div
    class="t-list t-size-m t-list--split"
  >
    <div
      class="t-list__header"
    >
      通过 Slot 插入的 Header
    </div>
    <ul
      class="t-list-items"
    >
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="t-list__footer"
    >
      通过 Slot 插入的 Footer
    </div>
    <div
      class="t-list__load"
    />
  </div>
</div>
`;

exports[`Button imageText demo works fine 1`] = `
<div>
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
   
  <div
    style="margin-bottom: 16px;"
  />
   
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                <p>
                  列表内容列表内容列表内容
                </p>
                 
                <p>
                  列表内容列表内容列表内容
                </p>
                 
                <p>
                  列表内容列表内容列表内容
                </p>
                 
                <p>
                  列表内容列表内容列表内容
                </p>
                 
                <p>
                  列表内容列表内容列表内容
                </p>
              </p>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
</div>
`;

exports[`Button multiline demo works fine 1`] = `
<div>
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
</div>
`;

exports[`Button operation demo works fine 1`] = `
<div>
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
          <li
            class="t-list-item__action"
          >
            <span>
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作1
                </span>
              </button>
               
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作2
                </span>
              </button>
               
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作3
                </span>
              </button>
            </span>
          </li>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
          <li
            class="t-list-item__action"
          >
            <span>
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作1
                </span>
              </button>
               
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作2
                </span>
              </button>
               
              <button
                class="t-button t-size-m t-button--variant-text t-button--theme-default"
                type="button"
              >
                <span
                  class="t-button__text"
                >
                  操作3
                </span>
              </button>
            </span>
          </li>
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
   
  <div
    style="margin-bottom: 16px;"
  />
   
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
          <li
            class="t-list-item__action"
          >
            <button
              class="t-button t-size-m t-button--variant-text t-button--theme-default t-button--shape-square"
              type="button"
            >
              <span
                class="t-button__text"
              >
                <svg
                  class="t-icon t-icon-edit"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M784.192 149.30624C772.6592 137.77088 757.8176 131.55904 742.72 130.672H735.1616C720.064 131.55904 705.2224 137.77088 693.6832 149.30624L580.54848 262.44352 761.568 443.4624 874.7072 330.32576C899.6992 305.33184 899.6992 264.8096 874.7072 239.81568L784.192 149.30624ZM535.29344 307.69792L716.3136 488.71744 309.01952 896.0128H128V714.9888L535.29344 307.69792ZM535.296 398.20864L191.936 741.504V832H282.496L625.792 488.70464 535.296 398.20864ZM738.9376 194.56064L671.04 262.40064 761.536 352.89664 829.4528 285.07072 738.9376 194.56064Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
             
            <button
              class="t-button t-size-m t-button--variant-text t-button--theme-default t-button--shape-square"
              type="button"
            >
              <span
                class="t-button__text"
              >
                <svg
                  class="t-icon t-icon-download"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M896 896L192 896 192 704 256 704 255.937328 831.9609152 831.9373312 831.9609152 832 704 896 704 896 896ZM576 75.4169184L575.937328 659.968 853.184352 381.09423936 897.6297088 425.53959744 565.92213248 758.7649856 563.20206976 761.182432 560.27266048 763.2049344 559.58821952 763.6247552C550.50648384 769.5321728 538.68397632 769.4512128 529.68399616 763.419968L525.4449728 759.908672 524.72661888 759.2139008 524.54518144 759.0378112 524.36488768 758.8598464 523.24472192 757.681312 191.873328 421.73694912 234.68549824 378.92477952 511.937328 662.144 512 75.4169184 576 75.4169184Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          <div
            class="t-list-item__meta"
          >
            <div
              class="t-list-item__meta-avatar"
            >
              <img
                src="https://tdesign.gtimg.com/list-icon.png"
              />
            </div>
            <div
              class="t-list-item__meta-content"
            >
              <h3
                class="t-list-item__meta-title"
              >
                列表主内容
              </h3>
              <p
                class="t-list-item__meta-description"
              >
                列表内容列表内容
              </p>
            </div>
          </div>
          <li
            class="t-list-item__action"
          >
            <button
              class="t-button t-size-m t-button--variant-text t-button--theme-default t-button--shape-square"
              type="button"
            >
              <span
                class="t-button__text"
              >
                <svg
                  class="t-icon t-icon-edit"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M784.192 149.30624C772.6592 137.77088 757.8176 131.55904 742.72 130.672H735.1616C720.064 131.55904 705.2224 137.77088 693.6832 149.30624L580.54848 262.44352 761.568 443.4624 874.7072 330.32576C899.6992 305.33184 899.6992 264.8096 874.7072 239.81568L784.192 149.30624ZM535.29344 307.69792L716.3136 488.71744 309.01952 896.0128H128V714.9888L535.29344 307.69792ZM535.296 398.20864L191.936 741.504V832H282.496L625.792 488.70464 535.296 398.20864ZM738.9376 194.56064L671.04 262.40064 761.536 352.89664 829.4528 285.07072 738.9376 194.56064Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
             
            <button
              class="t-button t-size-m t-button--variant-text t-button--theme-default t-button--shape-square"
              type="button"
            >
              <span
                class="t-button__text"
              >
                <svg
                  class="t-icon t-icon-download"
                  version="1.1"
                  viewBox="0 0 1024 1024"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M896 896L192 896 192 704 256 704 255.937328 831.9609152 831.9373312 831.9609152 832 704 896 704 896 896ZM576 75.4169184L575.937328 659.968 853.184352 381.09423936 897.6297088 425.53959744 565.92213248 758.7649856 563.20206976 761.182432 560.27266048 763.2049344 559.58821952 763.6247552C550.50648384 769.5321728 538.68397632 769.4512128 529.68399616 763.419968L525.4449728 759.908672 524.72661888 759.2139008 524.54518144 759.0378112 524.36488768 758.8598464 523.24472192 757.681312 191.873328 421.73694912 234.68549824 378.92477952 511.937328 662.144 512 75.4169184 576 75.4169184Z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </li>
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
</div>
`;

exports[`Button scroll demo works fine 1`] = `
<div
  class="t-list t-size-m"
  style="max-height: 200px;"
>
  <ul
    class="t-list-items"
  >
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
     
    <li
      class="t-list-item"
    >
      <div
        class="t-list-item-main"
      >
        列表内容列表内容
      </div>
    </li>
  </ul>
  <div
    class="t-list__load"
  />
</div>
`;

exports[`Button size demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="t-list t-size-s t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
   
  <div
    class="t-list t-size-m t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
   
  <div
    class="t-list t-size-l t-list--split"
  >
    <ul
      class="t-list-items"
    >
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
       
      <li
        class="t-list-item"
      >
        <div
          class="t-list-item-main"
        >
          列表内容列表内容
        </div>
      </li>
    </ul>
    <div
      class="t-list__load"
    />
  </div>
</div>
`;
